robot-server/robot_server/protocols/protocol_models.py

```diff
@@ -78,3 +78,5 @@ class Protocol(ResourceModel):
         ...,
         description="An analysis of how the protocol is expected to run.",
     )
+
+    protocol_key: Optional[str] = None
```

Comment:
Public facing model fields should be camel case. Let's also simplify it to `key`, because it's already on a `Protocol` response

```suggestion
    key: Optional[str] = None
```